Makefile

```diff
@@ -119,14 +105,55 @@ beforedepend:: \
   utils/config.ml utils/config_boot.ml utils/config_main.ml \
   utils/domainstate.ml utils/domainstate.mli
 
-ocamllex_PROGRAMS := $(addprefix lex/,ocamllex ocamllex.opt)
+ocamllex_PROGRAMS = $(addprefix lex/,ocamllex ocamllex.opt)
 
 ocamlyacc_PROGRAM = yacc/ocamlyacc
 
-PROGRAMS = expunge ocaml ocamlc ocamlc.opt ocamlnat ocamlopt ocamlopt.opt \
-  $(ocamllex_PROGRAMS) $(ocamlyacc_PROGRAM)
+TOOLS_TO_INSTALL = \
+  ocamldep ocamlprof ocamlcp ocamlmklib ocamlmktop ocamlobjinfo
+
+ifeq "$(NATIVE_COMPILER)" "true"
+TOOLS_TO_INSTALL += ocamloptp
+endif
+
+TOOLS = $(TOOLS_TO_INSTALL) ocamlcmt dumpobj primreq stripdebug cmpbyt
+
+TOOLS_PROGRAMS = $(addprefix tools/,$(TOOLS))
+
+TOOLS_MODULES = tools/profiling
+
+# C programs
+
+C_PROGRAMS = $(ocamlyacc_PROGRAM)
+
+$(foreach PROGRAM, $(C_PROGRAMS),\
+  $(eval $(call PROGRAM_SYNONYM,$(PROGRAM))))
+
+# OCaml programs that are compiled in both bytecode and native code
+
+OCAML_PROGRAMS = ocamlc ocamlopt lex/ocamllex $(TOOLS_PROGRAMS)
```

Comment:
`tools/ocamloptp` needs adding to this list (as it's used for the `clean` target) - this is the cause of the CI failure at the moment